pkgs/by-name/vc/vcpkg/package.nix

```diff
@@ -16,6 +17,14 @@ stdenvNoCC.mkDerivation (finalAttrs: {
     hash = "sha256-HMK3sebq/9TuxHQ75+5UIMvN09cPWmq7TFBBwRY4X7o=";
   };
 
+  patches = [
```

Comment:
I don't think applying patch to vcpkg ports are a good idea.

First, I'm not even sure this patch will hit mainstream and I want to keep this package as vanilla as possible. 

Second, those patches can be applied by user via [overlay ports](https://learn.microsoft.com/en-us/vcpkg/concepts/overlay-ports). If those patch are needed by one of your project, no need to patch the official package, just add it to your git until maintain accepts the patch.

Unless you have a good argument why we should patch vcpkg ports (and therefore breaking the built-in baseline) I would abstain to diverge from upstream.

Reply:
Uh-huh, the intent here actually wasn't to ship these patches to vcpkg' users, but to imgui users. The direct implementation would've been to `applyPatches` in `imgui/default.nix`, but the more scalable solution is to create a fixpoint. I used the `vcpkg` package as one.

I'll revert the vcpkg change

Reply:
I updated the description and I no longer touch `pkgs.vcpkg`. There's no need to synchronize any patches yet

Reply:
If those changes benefit users of imgui even for those who use vcpkg as their package managers, I would recommend also sending the patch to upstream vcpkg.

Reply:
The upstream PR is at https://github.com/microsoft/vcpkg/pull/38159, although probably not yet in shape that upstream would accept.